### ISSUE TITLE
Error when testing kafka stream based on kop

### ISSUE THREAD

**Issue by keyboardbobo** [streamnative/kop#1472]
**Describe the bug**
I can test WordCountDemo of kafka stream based on native kafka, but I get an error when testing based on kop: `persistent://tenant/ns/streams-plaintext-input is unknown yet during rebalance, please make sure they have been pre-created before starting the Streams application.`. When I tested `AdminClient.listTopics()` alone, I found that the returned topic list was empty. Is this a query that the `META `interface does not support ` topics=empty`?

**To Reproduce**
AdminClient.listTopics() got empty result

**Expected behavior**
AdminClient.listTopics() return all topics of the user

**Screenshots**

**Additional context**

**Comment by BewareMyPower** [comment]
Could you give more details about how to reproduce it?

Here are how I tried to reproduce your issue.

First I ran KoP 2.10.1.6 on Pulsar 2.10.1 standalone. Then create 3 topics via Kafka CLI (2.7.1) commands.

```bash
$ ./bin/kafka-console-producer.sh --bootstrap-server localhost:9092 --topic topic0
>hello
wor>ld
>
$ ./bin/kafka-console-producer.sh --bootstrap-server localhost:9092 --topic topic2
>wwww
>                                                                                                                                                                                                                                                           $ ./bin/kafka-console-producer.sh --bootstrap-server localhost:9092 --topic topic1
> hello
>
```

Then I ran the following Java code.

```java
        final var props = new Properties();
        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
        try (AdminClient client = AdminClient.create(props)) {
            final Map<String, TopicListing> map = client.listTopics().namesToListings().get();
            map.forEach((name, topicListing) -> {
                System.out.println(name + " => " + topicListing.name() + " isInternal: " + topicListing.isInternal());
            });
        } catch (ExecutionException | InterruptedException e) {
            throw new RuntimeException(e);
        }
```

```xml
    <dependency>
      <groupId>org.apache.kafka</groupId>
      <artifactId>kafka-clients</artifactId>
      <version>3.0.0</version>
    </dependency>
```

The outputs are:

```
topic1 => topic1 isInternal: false
topic2 => topic2 isInternal: false
topic0 => topic0 isInternal: false
```

**Comment by keyboardbobo** [comment]
@BewareMyPower KoP version is 2.9.2.24, pulsar 2.9.2, kafka-client 2.1.1. 
I have created 4 pulsar topics in the namespace chenjianbo-tenant/chenjianbo-ns, but none of them can be queried through the client code:

> Properties props = new Properties();
>         props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, "127.0.0.1:9092");
>         props.put("security.protocol","SASL_PLAINTEXT");
>         props.put("sasl.mechanism","PLAIN");
>         props.put("sasl.jaas.config","org.apache.kafka.common.security.plain.PlainLoginModule required username=\"chenjianbo-tenant/chenjianbo-ns\"" +
>                 " password=\"token:xxxx\";");
>         AdminClient admin = KafkaAdminClient.create(props);
>         ListTopicsResult listTopicsResult = admin.listTopics();
>         Collection<TopicListing> topicListings = listTopicsResult.listings().get();
>         System.out.println("topic size：" + topicListings.size());

the output is:

> topic size：0

**Comment by Demogorgon314** [comment]
Can you use pulsar admin to list the `chenjianbo-tenant/chenjianbo-ns` namespace topics?

**Comment by BewareMyPower** [comment]
What's your `kopAllowedNamespaces` config? See https://github.com/streamnative/kop/blob/master/docs/configuration.md#namespace

I think you should add `chenjianbo-tenant/chenjianbo-ns` to the configured namespace list.

**Comment by keyboardbobo** [comment]
> What's your `kopAllowedNamespaces` config? See https://github.com/streamnative/kop/blob/master/docs/configuration.md#namespace
> 
> I think you should add `chenjianbo-tenant/chenjianbo-ns` to the configured namespace list.

I have debugged the` io.streamnative.pulsar.handlers.kop.KafkaRequestHandler#getTopicsAsync `method and it doesn't go
         `if ((request.topics() == null) || (request.topics().isEmpty() && request.version() == 0)) `this branch, so the kopAllowedNamespaces parameter configuration is useless, right?

**Comment by BewareMyPower** [comment]
Yeah. I'm wondering how did you debug it? Could you show the debug logs of METADATA requests? For example,

> 2022-09-07T17:23:45,962+0800 [pulsar-ph-kafka-79-8] DEBUG io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder - [/127.0.0.1:60513] Received kafka cmd RequestHeader(apiKey=METADATA, apiVersion=6, clientId=adminclient-1, correlationId=5), the request content is: KafkaHeaderAndRequest(header=RequestHeader(apiKey=METADATA, apiVersion=6, clientId=adminclient-1, correlationId=5), request={topics=null,allow_auto_topic_creation=true}, remoteAddress=/127.0.0.1:60513)

You can see the `request.topics()` is null from **`request={topics=null`** in my local env. So it goes into the if block you mentioned and list all topics under `public/default` (the default `kopAllowedNamespaces` config) namespace.

**Comment by BewareMyPower** [comment]
I tested kafka-clients 2.1.1 just now.

> 2022-09-07T17:34:17,770+0800 [pulsar-ph-kafka-79-10] DEBUG io.streamnative.pulsar.handlers.kop.KafkaCommandDecoder - [/127.0.0.1:62042] Received kafka cmd RequestHeader(apiKey=METADATA, apiVersion=6, clientId=adminclient-1, correlationId=3), the request content is: KafkaHeaderAndRequest(header=RequestHeader(apiKey=METADATA, apiVersion=6, clientId=adminclient-1, correlationId=3), request={topics=null,allow_auto_topic_creation=true}, remoteAddress=/127.0.0.1:62042)

The `request.topics()` is also null and list topics succeeded as well. You can see (and paste/upload) your own debug logs for more details

**Comment by keyboardbobo** [comment]
@BewareMyPower I use the arthas tool to debug:

> [arthas@163155]$ watch io.streamnative.pulsar.handlers.kop.KafkaRequestHandler getTopicsAsync "{params[0]}" -x 2
> Press Q or Ctrl+C to abort.
> Affect(class-cnt:1 , method-cnt:1) cost in 162 ms.
> ts=2022-09-07 18:08:17; [cost=2.982383ms] result=@ArrayList[
>     @MetadataRequest[
>         TOPICS_KEY_NAME=@String[topics],
>         ALLOW_AUTO_TOPIC_CREATION_KEY_NAME=@String[allow_auto_topic_creation],
>         METADATA_REQUEST_V0=@Schema[{topics:ARRAY(STRING)}],
>         METADATA_REQUEST_V1=@Schema[{topics:ARRAY(STRING)}],
>         METADATA_REQUEST_V2=@Schema[{topics:ARRAY(STRING)}],
>         METADATA_REQUEST_V3=@Schema[{topics:ARRAY(STRING)}],
>         METADATA_REQUEST_V4=@Schema[{topics:ARRAY(STRING),allow_auto_topic_creation:BOOLEAN}],
>         METADATA_REQUEST_V5=@Schema[{topics:ARRAY(STRING),allow_auto_topic_creation:BOOLEAN}],
>         METADATA_REQUEST_V6=@Schema[{topics:ARRAY(STRING),allow_auto_topic_creation:BOOLEAN}],
>         **topics=@ArrayList[isEmpty=true;size=0]**,
>         allowAutoTopicCreation=@Boolean[true],
>         version=@Short[6],
>     ],
> ]

**Comment by keyboardbobo** [comment]
![C2C4411B-5F4A-4B0B-A719-783444634547](https://user-images.githubusercontent.com/9923278/188854267-b73cd684-1051-4eff-b2ed-847c5ef47c9f.png)

**Comment by keyboardbobo** [comment]
@BewareMyPower @Demogorgon314 Sorry, maybe I did something wrong, configured
`kopAllowedNamespaces=chenjianbo-tenant/chenjianbo-ns` is already available, thanks! Can this value be configured with wildcards (such as kopAllowedNamespaces=*) like kafka, so that all topics of the user can be automatically obtained?Because tenants and namespaces are dynamically created by users, it is not easy to modify system configuration files

**Comment by BewareMyPower** [comment]
> Can this value be configured with wildcards (such as kopAllowedNamespaces=*)

Maybe yes. I think we can only a new issue for this feature request.

**Comment by keyboardbobo** [comment]
resolve
